src/utils/rest-api.js

```diff
@@ -723,3 +723,41 @@ export function getLoadFlowParameters(studyUuid) {
             : response.text().then((text) => Promise.reject(text))
     );
 }
+
+function changeLineStatus(studyUuid, lineId, status) {
+    const changeLineStatusUrl =
+        getStudyUrl(studyUuid) +
+        '/network-modification/lines/' +
+        encodeURIComponent(lineId) +
+        '/status';
+    console.debug('%s with body: %s', changeLineStatusUrl, status);
+    return backendFetch(changeLineStatusUrl, { method: 'put', body: status });
+}
+
+export function lockoutLine(studyUuid, lineId) {
+    console.info('locking out line ' + lineId + ' ...');
+    return changeLineStatus(studyUuid, lineId, 'lockout');
+}
+
+export function tripLine(studyUuid, lineId) {
+    console.info('tripping line ' + lineId + ' ...');
+    return changeLineStatus(studyUuid, lineId, 'trip');
+}
+
+export function energiseLineEnd(studyUuid, lineId, lineEnd) {
+    console.info('energise line ' + lineId + ' end ' + lineEnd + ' ...');
+    return changeLineStatus(
+        studyUuid,
+        lineId,
+        lineEnd === 'ONE'
+            ? 'energiseEndOne'
+            : lineEnd === 'TWO'
+            ? 'energiseEndTwo'
+            : null
```

Comment:
control if  side is valid before (or use an enum) to avoid 415 error

Reply:
Same here, I don't think using an enum would prevent from mistakes.